### PR TITLE
Lsaholds

### PR DESCRIPTION
**Pull request by adamshire123**
#### What does this PR do?
Adds additional functionality to Primo Barcode lookup addon for ILLiad, allowing staff to place holds in Alma using ILLiad cilent

#### Helpful background context
When we migrated to Alma our Illiad addon for placing holds in Aleph
for LSA scan and deliver service was no longer functional. These updates
to this addon reenable that functionality.

#### How can a reviewer manually see the effects of these changes?
1. install illiad client and this addon
2. configure addon
3. create a scan and deliver request via Primo
4. locate request in Primo and use button on MIT LSA Holds tab to place a hold in alma
5. confirm that the hold was place successfully in alma

#### What are the relevant tickets?

[- https://mitlibraries.atlassian.net/browse/DIP-](https://mitlibraries.atlassian.net/browse/ENSY-47)

#### Includes new or updated dependencies?

NO

#### Changes expectations for external applications?

NO

#### Developer

- [ ] All new config values are documented in README
- [ ] All new config values have been added to staging and production environments
- [ ] All related Jira tickets are linked in commit message(s)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

### CODE REVIEW

**Comment by adamshire123** [comment]
@JPrevost I think this ought to be reviewed by someone before it goes into production. This is the Lua work I mentioned during the lightning talks, which is why I'm picking on you. happy to zoom or slack about it if that's easier for you.

**Comment by JPrevost** [comment]
@adamshire123 do you have any idea why it's flagging every line as changed? It's pretty difficult to understand what is actually changing and what already existed the way it's flagging everything as a change in the `.lua` files

**Comment by JPrevost** [comment]
@adamshire123 I was able to suppress whitespace changes in the diff to see a cleaner change set. I should be able to look at this tomorrow, although I'm not entirely sure I'll have much useful input :)

File: .github/pull-request-template.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+#### How can a reviewer manually see the effects of these changes?
+
+Explain how to see the proposed changes in the application if possible.
+
+Delete this section if it isn't applicable to the PR.
+
+#### What are the relevant tickets?
+
+- https://mitlibraries.atlassian.net/browse/DIP-
```
Comment (JPrevost) [inline .github/pull-request-template.md]: Not a big deal, but I'd remove the `DIP` bit here and replace with the project slug this repo typically works out of... if there isn't a typical project in Jira for this, removing the starter may be best.

**Review by JPrevost — APPROVED** [review]
I wasn't able to verify any of this, but the general flow of the code looks fine.

In the future, I'd suggest style changes should happen in separate PRs as there was a lot of noise in this one with the changes to whitespace and semicolons as it would have been much less mental focus required to review the changes if they were separate (i.e. just make the whitespace/semicolon changes first then merge that then add the new features would have been easier to review by a lot).

If this works and has stakeholder approval of doing what is required, 👍🏻 

This is definitely interesting to be able to modify staff workflows in a vended solution like this. I hope it ends up being easy to maintain as it seems really useful.

**Comment by adamshire123** [comment]
Thanks @JPrevost . I was surprised by all of the white space changes, most of them were not intentional. I did remove all of the unnecessary semicolons. I hoped that keeping that in its own commit would have helped, but it still isn't that clear, so I see how your suggestion of a totally separate PR for that kind of cleanup makes more sense.

**Pull request merged**